lib/src/command/token_add.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: import_of_legacy_library_into_null_safe
+
+import 'dart:async';
+
+import '../authentication/token.dart';
+import '../command.dart';
+import '../io.dart';
+import '../log.dart' as log;
+import '../source/hosted.dart';
+
+/// Handles the `token add` pub command.
+class TokenAddCommand extends PubCommand {
+  @override
+  String get name => 'add';
+  @override
+  String get description => 'Add token for a server.';
+  @override
+  String get invocation => 'pub token add';
+
+  @override
+  Future<void> runProtected() async {
+    if (argResults.rest.isEmpty) {
+      usageException('Must specify a hosted server URL to be added.');
+    } else if (argResults.rest.length > 1) {
+      usageException('Takes only a single argument.');
+    }
+
+    try {
+      var hostedUrl = validateAndNormalizeHostedUrl(argResults.rest.first);
+
+      if (hostedUrl.isScheme('HTTP')) {
+        // TODO(themisir): Improve the following message.
+        usageException('Unsecure pub server could not be added.');
```

Comment:
This is not a `usageException`, you are using the command-line arguments correctly, but we are forbidding you from doing what you're trying to do.

Maybe a `DataException` is more appropriate :D